pkg/tagger/collectors/ecs_main.go

```diff
@@ -63,14 +78,22 @@ func (c *ECSCollector) Fetch(entity string) ([]string, []string, []string, error
 		return nil, nil, nil, nil
 	}
 
-	tasks_list, err := c.ecsUtil.GetTasks()
+	tasks, err := c.metaV1.GetTasks()
 	if err != nil {
 		return []string{}, []string{}, []string{}, err
 	}
-	updates, err := c.parseTasks(tasks_list, cID)
+
+	var updates []*TagInfo
+
+	if config.Datadog.GetBool("ecs_collect_resource_tags_ec2") && ecsutil.HasEC2ResourceTags() {
```

Comment:
I'm wondering if we can put this test in `addTagsForContainer ` and always call `updates, err = c.parseTasks(tasks, cID, addTagsForContainer)`

Reply:
I thought that since in this case this test will be executed for each container in each task, it could lead to strange results if there's an issue updating the cache key behind `HasEC2ResourceTags` and it may be more convenient to eval this beforehand then.